roles/redpanda_broker/tasks/install-redpanda.yml

```diff
@@ -1,17 +1,79 @@
 ---
-- name: Download the Redpanda repo script
-  ansible.builtin.get_url:
-    url: "{{ redpanda_repo_script }}"
-    dest: "/tmp/redpanda_repo_script.sh"
-    mode: '0755'
+- name: Download and import Redpanda GPG Key
+  shell: |
+    curl -1sLf "{{ rp_key_deb }}" | gpg --dearmor > {{ rp_key_path_deb }}
+  when: ansible_os_family == 'Debian'
+
+- name:  Add Redpanda DEB repository
+  apt_repository:
+    filename: "redpanda-redpanda"
+    mode: 0644
+    repo:  "{{ rp_repo_signing_deb }}"
+    update_cache: yes
+  when: ansible_os_family == 'Debian'
 
-- name: Execute the Redpanda repo script
-  ansible.builtin.command:
-    cmd: sudo -E /tmp/redpanda_repo_script.sh
-  changed_when: false
+- name: Add Redpanda DEB repository (source)
+  apt_repository:
+    filename: "redpanda-redpanda-src"
+    mode: 0644
+    repo: "{{ rp_repo_signing_src_deb }}"
+    update_cache: yes
+  when: ansible_os_family == 'Debian'
+
+
+- name: Add redpanda-redpanda RPM repository
+  ansible.builtin.yum_repository:
+    name: 'redpanda-redpanda'
+    description: 'redpanda-redpanda'
+    baseurl: "{{ redpanda_base_url }}/public/redpanda/rpm/{{ ansible_distribution | lower }}/{{ ansible_distribution_major_version }}/$basearch"
+    gpgkey: "{{ rp_key_rpm }}"
+    repo_gpgcheck: yes
+    gpgcheck: yes
+    enabled: yes
+    sslcacert: '/etc/pki/tls/certs/ca-bundle.crt'
+    sslverify: yes
+    metadata_expire: 300
+    skip_if_unavailable: yes
+  when: ansible_os_family == 'RedHat'
+
+- name: Add redpanda-redpanda-noarch RPM repository
+  ansible.builtin.yum_repository:
+    name: 'redpanda-redpanda-noarch'
+    description: 'redpanda-redpanda-noarch'
+    baseurl: "{{ redpanda_base_url }}/public/redpanda/rpm/{{ ansible_distribution | lower }}/{{ ansible_distribution_major_version }}/$basearch"
+    gpgkey: "{{ rp_key_rpm }}"
+    repo_gpgcheck: yes
+    gpgcheck: yes
+    enabled: yes
+    sslcacert: '/etc/pki/tls/certs/ca-bundle.crt'
+    sslverify: yes
+    metadata_expire: 300
+    skip_if_unavailable: yes
+  when: ansible_os_family == 'RedHat'
+
+- name: Add redpanda-redpanda-source RPM repository
+  ansible.builtin.yum_repository:
+    name: 'redpanda-redpanda-source'
+    description: 'redpanda-redpanda-source'
+    baseurl: "{{ redpanda_base_url }}/public/redpanda/rpm/{{ ansible_distribution | lower }}/{{ ansible_distribution_major_version }}/$basearch"
+    gpgkey: "{{ rp_key_rpm }}"
+    repo_gpgcheck: yes
+    gpgcheck: yes
+    enabled: yes
+    sslcacert: '/etc/pki/tls/certs/ca-bundle.crt'
+    sslverify: yes
+    metadata_expire: 300
+    skip_if_unavailable: yes
+  when: ansible_os_family == 'RedHat'
+
+- name: Install GPG key (RPM)
+  ansible.builtin.rpm_key:
+    state: present
+    key: "{{ rp_key_rpm }}"
+  when: ansible_os_family == 'RedHat'
 
 - name: Install redpanda from repository
-  ansible.builtin.apt:
+  ansible.builtin.package:
```

Comment:
Why the change? I'm curious.

`ansible.builtin.package` defaults to **use:auto** and detects which package manager should use, but we have a **when** clause checking for the OS family. Wonder if we are using this `package` maybe we can migrate both (L76 and L85)

Reply:
I had intended to convert the name block to accept a fact + eliminate the duplicate .package call but never got around to the fact.

Reply:
done